### PR TITLE
networkanalytics with modular

### PR DESCRIPTION
**Pull request by kazrael2119**
fixes https://github.com/Azure/autorest.typescript/issues/2535
fixes https://github.com/Azure/autorest.typescript/issues/2353

### CODE REVIEW

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/rest/models.ts [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+export interface ManagedServiceIdentity {
+  /** The type of managed identity assigned to this resource. */
+  type: ManagedServiceIdentityType;
+  /** The identities assigned to this resource by the user. */
+  userAssignedIdentities?: UserAssignedIdentities;
+}
+
+/** The set of user assigned identities associated with the resource. The userAssignedIdentities dictionary keys will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}. The dictionary values can be empty objects ({}) in requests.", */
+export interface UserAssignedIdentities
+  extends Record<string, UserAssignedIdentity> {}
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/rest/models.ts:158]: Here is the RLC generation.

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/models/models.ts [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+/** The properties of the managed service identities assigned to this resource. */
+export interface ManagedServiceIdentity {
+  /** The Active Directory tenant id of the principal. */
+  readonly tenantId?: string;
+  /** The active directory identifier of this principal. */
+  readonly principalId?: string;
+  /** The type of managed identity assigned to this resource. */
+  type: ManagedServiceIdentityType;
+  /** The identities assigned to this resource by the user. */
+  userAssignedIdentities?: Record<string, UserAssignedIdentity>;
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/models/models.ts:180]: The modular would be generated as Record.

Comment (qiaozha) [reply packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/models/models.ts:180]: we should generate the same as RLC, I think it's a bug.

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 3]
```diff
@@ -306,6 +306,9 @@ function getType(
   options: { disableEffectiveModel?: boolean; usage?: UsageFlags } = {}
 ): any {
   // don't cache simple type(string, int, etc) since decorators may change the result
+  if (type.kind === "Model" && type.name === "ManagedServiceIdentity") {
+    type;
+  }
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:311]:
```suggestion
```

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 4]
```diff
@@ -306,6 +306,9 @@ function getType(
   options: { disableEffectiveModel?: boolean; usage?: UsageFlags } = {}
 ): any {
   // don't cache simple type(string, int, etc) since decorators may change the result
+  if (type.kind === "Model" && type.name === "ManagedServiceIdentity") {
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:309]: debugging code?

File: packages/typespec-ts/src/modular/emitModels.ts [hunk 5]
```diff
@@ -195,6 +195,9 @@ export function buildModels(
       const enumAlias = buildEnumModel(model);
       modelsFile.addTypeAlias(enumAlias);
     } else {
+      if (model.name === "UserAssignedIdentities") {
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/emitModels.ts:198]: debugging code?

File: packages/typespec-ts/src/modular/emitModels.ts [hunk 6]
```diff
@@ -195,6 +195,9 @@ export function buildModels(
...
+        model;
+      }
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/emitModels.ts:200]:
```suggestion
```

File: packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/rest/networkAnalyticsClient.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+ * Initialize a new instance of `NetworkAnalyticsContext`
+ * @param credentials - uniquely identify client credential
+ * @param options - the parameter for all optional parameters
+ */
+export default function createClient(
+  credentials: TokenCredential,
+  options: ClientOptions = {},
+): NetworkAnalyticsContext {
+  const endpointUrl =
+    options.endpoint ?? options.baseUrl ?? `https://management.azure.com`;
```
Comment (MaryGao) [inline packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/rest/networkAnalyticsClient.ts:19]: this issue is fixed in latest generation https://github.com/Azure/autorest.typescript/issues/2349

**Review by qiaozha — APPROVED** [review]

File: packages/typespec-test/test/openai_generic/generated/typespec-ts/src/models/models.ts [hunk 8]
```diff
@@ -770,9 +770,12 @@ export interface ChatCompletionFunctions {
    * about the format.\n\nTo describe a function that accepts no parameters, provide the value
    * `{\"type\": \"object\", \"properties\": {}}`.
    */
-  parameters: Record<string, unknown>;
+  parameters: ChatCompletionFunctionParameters;
 }
 
+export interface ChatCompletionFunctionParameters
+  extends Record<string, unknown> {}
```
Comment (qiaozha) [inline packages/typespec-test/test/openai_generic/generated/typespec-ts/src/models/models.ts:777]: we should treat unknown as any?

Comment (qiaozha) [reply packages/typespec-test/test/openai_generic/generated/typespec-ts/src/models/models.ts:777]: use https://github.com/Azure/autorest.typescript/issues/2078 to track it.

**Pull request merged**